moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -116,11 +117,23 @@ void ompl_interface::ModelBasedPlanningContext::configure(const ros::NodeHandle&
   ompl_simple_setup_->getStateSpace()->setStateSamplerAllocator(
       std::bind(&ModelBasedPlanningContext::allocPathConstrainedSampler, this, std::placeholders::_1));
 
-  // convert the input state to the corresponding OMPL state
-  ompl::base::ScopedState<> ompl_start_state(spec_.state_space_);
-  spec_.state_space_->copyToOMPLState(ompl_start_state.get(), getCompleteInitialRobotState());
-  ompl_simple_setup_->setStartState(ompl_start_state);
-  ompl_simple_setup_->setStateValidityChecker(ob::StateValidityCheckerPtr(new StateValidityChecker(this)));
+  if (spec_.constrained_state_space_)
+  {
+    // convert the input state to the corresponding OMPL state
+    ompl::base::ScopedState<> ompl_start_state(spec_.constrained_state_space_);
+    spec_.state_space_->copyToOMPLState(ompl_start_state.get(), getCompleteInitialRobotState());
+    ompl_simple_setup_->setStartState(ompl_start_state);
+    ompl_simple_setup_->setStateValidityChecker(
+        ob::StateValidityCheckerPtr(new ConstrainedPlanningStateValidityChecker(this)));
+  }
+  else
+  {
+    // convert the input state to the corresponding OMPL state
+    ompl::base::ScopedState<> ompl_start_state(spec_.state_space_);
+    spec_.state_space_->copyToOMPLState(ompl_start_state.get(), getCompleteInitialRobotState());
+    ompl_simple_setup_->setStartState(ompl_start_state);
+    ompl_simple_setup_->setStateValidityChecker(ob::StateValidityCheckerPtr(new StateValidityChecker(this)));
```

Comment:
Here too with the shared_ptr.  Please do a search through your pr for the keyword `new` and try to remove it in any place it could be a make_shared or make_unique.

Reply:
I estimating there are close to 30 instances of this in the existing code base, but I think it will be confusing to change them all in this pull request. I changed these two instances to make_shared. But I'm not sure make_unique is ok here.